pkg/cache/rediscache/hashcache.go

```diff
@@ -68,7 +68,22 @@ func (r *RedisHashCache) Put(k string, v interface{}) error {
 	conn := r.redis.Get()
 	defer conn.Close()
 	_, err := conn.Do("HSET", r.key, k, v)
-	if r.ttl != 0 {
+	if err != nil {
+		return err
+	}
+
+	// Skip set TTL if unnecessary.
+	if r.ttl == 0 {
+		return nil
+	}
+
+	rep, err := redigo.Int(conn.Do("TTL", r.key))
+	if err != nil {
+		return err
+	}
+	// Only set TTL for hashkey in case TTL command return key has no TTL.
+	// ref: https://redis.io/commands/TTL
+	if rep < 0 {
```

Comment:
Could you add a comment about the TTL behavior for `NewTTLHashCache` function?
Because that TTL will be applied for the whole key instead of each individual field of the key.

Reply:
Addressed by https://github.com/pipe-cd/pipe/pull/2235/commits/e1e2a2722130dd35367c576ab50a3ce2de075eb6 🙏